compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -1048,8 +1048,9 @@ abstract class GenJSCode extends plugins.PluginComponent
         if (scalaPrimitives.isPrimitive(sym) &&
             !jsPrimitives.shouldEmitPrimitiveBody(sym)) {
           None
-        } else if (sym.isDeferred || sym.owner.isInterface) {
-          val body = if (sym.hasAnnotation(JavaDefaultMethodAnnotation)) {
+        } else if (rhs == EmptyTree) {
+          val body = if (scalaUsesImplClasses &&
```

Comment:
Hmmm... I guess you can keep it. I was trying to lure you into getting rid of `scalaUsesImplClasses`. But the other usage is really hard to replace :P